### DISCUSSION TITLE
ros2idl support

### DISCUSSION THREAD

**Discussion by MaxPRon** [foxglove/mcap#1095] [General]
Hello everyone,

I'm trying to read from a specific topic with wit ros2idl defined message. Here is a the slightly modified example:

`
def read_all_messages(filename,topics):

    messages = []

    with open(filename, "rb") as f:
        reader = make_reader(f)
        decoder = Decoder()

        for schema, channel, message in reader.iter_messages(topics=topics):
            messages.append(decoder.decode(schema, message))

    return messages`

However I always get the following message error:

`Exception has occurred: McapROS2DecodeError
can't parse schema with encoding "ros2idl"`

Is there something I'm missing or is this not yet supported ?

Best regards and thanks a lot already,

Max

**Comment by defunctzombie** [comment]
I don't recall that we've added support for `ros2idl` to the python ros2 mcap support package. If you have a `ros2idl` mcap file you created using ROS I'd recommend using the ROS python libraries to read it.

**Reply by MaxPRon** [reply]
Thanks for the answer. This means it's currently only possible with the respective ros2 installation meaning all message definitions installed etc. ?

**Reply by defunctzombie** [reply]
I believe that's the situation for reading ros2idl messages but could be wrong on that and there might be a way without a full install of all messages.

For anyone keen to explore adding support to read these without a full ROS2 install - the place to do that for mcap would be here: https://github.com/foxglove/mcap/tree/main/python/mcap-ros2-support

cc @emersonknapp

**Reply by MaxPRon** [reply]
@defunctzombie Thanks! There was an announcement recently regarding foxglove studio: https://foxglove.dev/blog/announcing-ros2-idl-support-in-foxglove-studio

It might hasn't made it to the helper functions yet. But I'll see what I can figure out
